llama-index-integrations/indices/llama-index-indices-managed-llama-cloud/llama_index/indices/managed/llama_cloud/base.py

```diff
@@ -57,38 +62,57 @@ class LlamaCloudIndex(BaseManagedIndex):
 
     def __init__(
         self,
-        name: str,
-        nodes: Optional[List[BaseNode]] = None,
-        transformations: Optional[List[TransformComponent]] = None,
-        timeout: int = 60,
+        # index identifier
+        name: Optional[str] = None,
+        pipeline_id: Optional[str] = None,
+        index_id: Optional[str] = None,  # alias for pipeline_id
+        id: Optional[str] = None,  # alias for pipeline_id
+        # project identifier
         project_name: str = DEFAULT_PROJECT_NAME,
         organization_id: Optional[str] = None,
+        # connection params
         api_key: Optional[str] = None,
         base_url: Optional[str] = None,
         app_url: Optional[str] = None,
-        show_progress: bool = False,
-        callback_manager: Optional[CallbackManager] = None,
+        timeout: int = 60,
         httpx_client: Optional[httpx.Client] = None,
         async_httpx_client: Optional[httpx.AsyncClient] = None,
+        # misc
+        show_progress: bool = False,
+        callback_manager: Optional[CallbackManager] = None,
+        # deprecated
+        nodes: Optional[List[BaseNode]] = None,
+        transformations: Optional[List[TransformComponent]] = None,
         **kwargs: Any,
     ) -> None:
         """Initialize the Platform Index."""
         self.name = name
         self.project_name = project_name
         self.organization_id = organization_id
-        self.transformations = transformations or []
+        self._pipeline_id = id or index_id or pipeline_id
+
+        if self._pipeline_id is None and self.name is None:
+            raise ValueError("One of index_id, pipeline_id or name must be provided")
 
         if nodes is not None:
             # TODO: How to handle uploading nodes without running transforms on them?
             raise ValueError("LlamaCloudIndex does not support nodes on initialization")
 
+        if transformations is not None:
+            raise ValueError(
+                "Setting transformations is deprecated for LlamaCloudIndex"
```

Comment:
Can we suggest what the alternative is here?

Reply:
good call.